dashboard/client/src/pages/serve/ServeReplicaDetailPage.tsx

```diff
@@ -0,0 +1,165 @@
+      <MainNavPageInfo
+        pageInfo={{
+          id: "serveDeployentDetail",
+          title: deployment.name,
+        }}
+      />
+      <MainNavPageInfo
+        pageInfo={{
+          id: "serveReplicaDetail",
+          title: replica_id,
+          path: `/serve/applications/${encodeURIComponent(
+            appName,
+          )}/${encodeURIComponent(deployment.name)}/${encodeURIComponent(
+            replica_id,
+          )}`,
+        }}
+      />
+      <MetadataSection
+        metadataList={[
+          {
+            label: "ID",
+            content: {
+              value: replica_id,
+              copyableValue: replica_id,
+            },
+          },
+          {
+            label: "State",
+            content: <StatusChip type="serveReplica" status={state} />,
+          },
+          {
+            label: "Logs",
+            content: (
+              <ServeReplicaLogsLink replica={replica} deployment={deployment} />
+            ),
+          },
+          {
+            label: "Actor ID",
+            content: {
+              value: actor_id ? actor_id : "-",
+              copyableValue: actor_id ? actor_id : undefined,
+            },
+          },
+          {
+            label: "Actor name",
+            content: {
+              value: actor_name,
+              copyableValue: actor_name,
+            },
+          },
+          {
+            label: "Node ID",
+            content: {
+              value: node_id ? node_id : "-",
+              copyableValue: node_id ? node_id : undefined,
+            },
+          },
+          {
+            label: "Node IP",
+            content: {
+              value: node_ip ? node_ip : "-",
+              copyableValue: node_ip ? node_ip : undefined,
+            },
+          },
+          {
+            label: "PID",
+            content: {
+              value: pid ? pid : "-",
+              copyableValue: pid ? pid : undefined,
+            },
+          },
+          {
+            label: "Started at",
+            content: {
+              value: formatDateFromTimeMs(start_time_s * 1000),
+            },
+          },
+          {
+            label: "Duration",
+            content: <DurationText startTime={start_time_s * 1000} />,
+          },
+        ]}
+      />
+    </div>
+  );
+};
+
+export type ServeReplicaLogsLinkProps = {
+  replica: ServeReplica;
+  deployment: ServeDeployment;
+};
+
+export const ServeReplicaLogsLink = ({
+  replica: { replica_id, node_ip },
+  deployment: { name: deploymentName },
+}: ServeReplicaLogsLinkProps) => {
+  const { ipLogMap } = useContext(GlobalContext);
+
+  let link: string | undefined;
+
+  if (node_ip && ipLogMap[node_ip]) {
+    // TODO(aguo): Have API return the location of the logs.
+    const path = `/serve/deployment_${deploymentName}_${replica_id}.log`;
+    link = `/logs/${encodeURIComponent(ipLogMap[node_ip])}/${encodeURIComponent(
+      path,
+    )}`;
```

Comment:
@zcin @edoakes can we make this TODO happen?

Right now the frontend assumes the location of the log files. If API tells us it seems less likely that the log location will change without updating the relevant API code.

Reply:
That should be pretty straightforward. @zcin we can add some logic to fetch this from the logger and tack it onto the metadata we return from the replica.